base/src/commonTest/kotlin/utils/timer/TimerTest.kt

```diff
@@ -0,0 +1,140 @@
+/*
+ Copyright 2021 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package utils.timer
+
+import com.splendo.kaluga.base.runBlocking
+import com.splendo.kaluga.base.utils.timer.Timer
+import com.splendo.kaluga.base.utils.timer.TimerProvider
+import com.splendo.kaluga.base.utils.timer.awaitFinish
+import com.splendo.kaluga.base.utils.timer.elapsed
+import kotlinx.coroutines.TimeoutCancellationException
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.withTimeout
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+import kotlin.time.Duration
+import kotlin.time.ExperimentalTime
+
+@ExperimentalTime
+class TimerTest {
```

Comment:
I think it's better to subclass from BaseFlowTest (also see FlowTest and SimpleFlowTest)
https://github.com/splendo/kaluga/blob/8e8fe0bcdb7ea4dc683f7cc4d412cce8f2717bce/test-utils/src/commonMain/kotlin/FlowTest.kt#L95

Then you can use `testWithFlow`, `action` and `test`.

Reply:
then probably I'd need to create a separate module, because it's currently in `base`. seemed like an overkill at the time of creation...

Reply:
A separate module is a good idea, not sure why this is in base.

note there is also a ticket to move the date stuff out for example, maybe they can share a module.

Reply:
extracted a separate module